cmake/Modules_CUDA_fix/upstream/FindCUDA/select_compute_arch.cmake

```diff
@@ -82,9 +82,15 @@ endif()
 if(CUDA_VERSION VERSION_GREATER "12.6")
   list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES "Blackwell")
   list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "10.0")
+  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "10.0a")
+  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "10.1a")
   list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "12.0")
+  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "12.0a")
```

Comment:
Are you sure if 12.6 will support 12.0a? Or 12.8 is needed for this one? 
Also, building everything with 12.0a might be an overkill as it affects probably only 1-2 operators (flex attention for example), as we are already doing

Reply:
> Are you sure if 12.6 will support 12.0a? Or 12.8 is needed for this one? Also, building everything with 12.0a might be an overkill as it affects probably only 1-2 operators (flex attention for example), as we are already doing

all these new blackwell needs, the new cuda 12.8

Reply:
![image](https://github.com/user-attachments/assets/46b43678-c547-424f-8622-75eba4ce3ee7)

Reply:
> Are you sure if 12.6 will support 12.0a? Or 12.8 is needed for this one? Also, building everything with 12.0a might be an overkill as it affects probably only 1-2 operators (flex attention for example), as we are already doing

up to you!

Reply:
Also when I updated this the name 12.8 wasn't released at the time hence I had this 'greater than' 12.6. I think thats fine to keep as is

Reply:
@drisspg are those architectures will be selected by default when user builds with 12.8? Or only for few ops? There are no difference between 12.0 and 12.0a for matmul or anything like that is there?

Reply:
So nothing within pytorch uses the 10 + 'a' features today. We do have one file that builds with sm90a for which is the rowwise kernel. 

So if building form source My understanding is that if you dont set cuda-arch list we warn saying you should. We then try auto: https://github.com/pytorch/pytorch/blob/a57133e3c7791bb61820274168e2666f397952ea/cmake/Modules_CUDA_fix/upstream/FindCUDA/select_compute_arch.cmake#L170
to detect w/ gpu you are on

otherwise yah we we would fall back to the above which adds them all.

I think its fine either way, you can build today from source w/ 10a if you set it manually in arch list.

The cpp_extensions though needs to be updated since that errors if you aren't in that list